src/components/LHNOptionsList/LHNOptionsList.tsx

```diff
@@ -0,0 +1,120 @@
+    style,
+    contentContainerStyles,
+    data,
+    onSelectRow,
+    optionMode,
+    shouldDisableFocusOptions = false,
+    reports = {},
+    reportActions = {},
+    policy = {},
+    preferredLocale = CONST.LOCALES.DEFAULT,
+    personalDetails = {},
+    transactions = {},
+    currentReportID = '',
+    draftComments = {},
+}: LHNOptionsListProps) {
+    const styles = useThemeStyles();
+    /**
+     * Function which renders a row in the list
+     */
+    const renderItem = useCallback(
+        ({item: reportID}: RenderItemProps): ReactElement => {
+            const itemFullReport = reports?.[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] ?? null;
+            const itemReportActions = reportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`] ?? null;
+            const itemParentReportActions = reportActions?.[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${itemFullReport?.parentReportID}`] ?? null;
+            const itemParentReportAction = itemParentReportActions?.[itemFullReport?.parentReportActionID ?? ''] ?? null;
+            const itemPolicy = policy?.[`${ONYXKEYS.COLLECTION.POLICY}${itemFullReport?.policyID}`] ?? null;
+            const transactionID = itemParentReportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU ? itemParentReportAction.originalMessage.IOUTransactionID ?? '' : '';
+            const itemTransaction = transactions?.[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`] ?? null;
+            const itemComment = draftComments?.[`${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${reportID}`] ?? '';
+            const participants = [...ReportUtils.getParticipantsIDs(itemFullReport), itemFullReport?.ownerAccountID];
+
+            const participantsPersonalDetails = OptionsListUtils.getPersonalDetailsForAccountIDs(participants, personalDetails);
+            return (
+                <OptionRowLHNData
+                    reportID={reportID}
+                    fullReport={itemFullReport}
+                    reportActions={itemReportActions}
+                    parentReportAction={itemParentReportAction}
+                    policy={itemPolicy}
+                    // @ts-expect-error TODO: Remove this once OptionsListUtils (https://github.com/Expensify/App/issues/24921) is migrated to TypeScript.
+                    personalDetails={participantsPersonalDetails}
+                    transaction={itemTransaction}
+                    receiptTransactions={transactions}
+                    viewMode={optionMode}
+                    isFocused={!shouldDisableFocusOptions && reportID === currentReportID}
+                    onSelectRow={onSelectRow}
+                    preferredLocale={preferredLocale ?? CONST.LOCALES.DEFAULT}
+                    comment={itemComment}
+                />
+            );
+        },
+        [currentReportID, draftComments, onSelectRow, optionMode, personalDetails, policy, preferredLocale, reportActions, reports, shouldDisableFocusOptions, transactions],
```

Comment:
Looks like this deps. array uses a lot of properties. Do we really feel benefits of this `useCallback`?

Reply:
I see your concern but I think this shouldn't be refactored or anything during ts migration

Reply:
Got it.